ProcessMaker/Models/ProcessRequest.php

```diff
@@ -20,6 +20,7 @@
 use ProcessMaker\Nayra\Engine\ExecutionInstanceTrait;
 use ProcessMaker\Repositories\BpmnDocument;
 use ProcessMaker\Traits\ExtendedPMQL;
+use ProcessMaker\Traits\ForUserScope;
```

Comment:
Maybe this is not part of the PR, but have you considered to add the following indexes to improve the subqueries:

- Index by `(process_request_id, is_self_service, self_service_groups columns)` for `process_request_tokens` table, to improve the subquery for self-service
- Index by `(process_request_id, user_id) for `process_request_tokens` table, to improve the subquery for participated (note that the currently existing for user_id does not contain process_request_id)

Reply:
@caleeli I think this should be a separate issue since the client that requested this fix tested it out on their test instance and approved it as is. As requests grow, we'll probably need more optimization like this in the near future.

Reply:
Agree